### PR TITLE
bug fix + tests

### PR DESCRIPTION
**Pull request by ltyson**
Incorporates this [original fix](https://github.com/lightstep/terraform-provider-lightstep/pull/185) from @adcharre (which eloquently describes the bug), adds a regression test for it, and bumps the provider version.

### CODE REVIEW

File: lightstep/resource_alert_test.go [hunk 1]
```diff
@@ -123,14 +123,14 @@ resource "lightstep_alert" "test" {
 			{
 				Config: badAlertMissingQueryAndCompositeFields,
 				Check: resource.ComposeTestCheckFunc(
-					testAccChecLightstepAlertExists(resourceName, &condition),
+					testAccCheckLightstepAlertExists(resourceName, &condition),
```
Comment (ltyson) [inline lightstep/resource_alert_test.go:126]: there was a typo in the name of this function and it was bugging me :)

**Review by mattcarmody — APPROVED** [review]
LGTM 🎉

File: examples/versions.tf [hunk 2]
```diff
@@ -7,7 +7,7 @@ terraform {
   required_providers {
     lightstep = {
       source  = "lightstep/lightstep"
-      version = "~> 1.61.1"
+      version = "~> 1.61.2"
```
Comment (ltyson) [inline examples/versions.tf:10]: oops, I changed the wrong file to update the version number 🤦‍♀️ this was totally not it

Comment (ltyson) [reply examples/versions.tf:10]: Here's the follow-up: https://github.com/lightstep/terraform-provider-lightstep/pull/190

**Pull request merged**